vms/administrator/tests/test_formFields.py

```diff
@@ -77,170 +80,159 @@ def login_admin(self):
-
-        self.assertEqual(settings.get_event_name_error(),'This field is required.')
-        self.assertEqual(settings.get_event_start_date_error(),'This field is required.')
-        self.assertEqual(settings.get_event_end_date_error(),'This field is required.')
+        # Checks:
+        # Event not edited
+        # Error messages appear
+        self.assertNotEqual(self.driver.current_url, self.live_server_url + settings.event_list_page)
+        self.assertEqual(len(settings.get_help_blocks()), 3)
+        self.assertEqual(settings.get_event_name_error(), 'This field is required.')
+        self.assertEqual(settings.get_event_start_date_error(), 'This field is required.')
+        self.assertEqual(settings.get_event_end_date_error(), 'This field is required.')
 
-        # database check to ensure that event not edited
+        # Check Database for edits
         self.assertEqual(len(Event.objects.all()), 1)
         self.assertEqual(len(Event.objects.filter(name=edited_event[0])), 0)
 
     def test_null_values_in_create_job(self):
-
-        # register event first to create job
-        event = ['event-name', '2017-08-21', '2017-09-28']
+        # Register Event
+        event = ['event-name', '2050-05-24', '2050-05-28']
         created_event = create_event_with_details(event)
 
-        # create job with null values
-        job = [created_event.id, '', '', '', '']
-
+        self.settings.go_to_events_page()
         settings = self.settings
         settings.live_server_url = self.live_server_url
+
+        # Create Job of null values
+        job = [created_event.id, '', '', '', '']
+        settings.navigate_to_job_list_view()
         settings.go_to_create_job_page()
         settings.fill_job_form(job)
 
-        # check that job was not created and that error messages appear as
-        # expected
-        self.assertEqual(self.driver.current_url,
-                         self.live_server_url + settings.create_job_page)
+        # Checks:
+        # Job not created
+        # Error messages appear
+        self.assertEqual(settings.remove_i18n(self.driver.current_url), self.live_server_url + settings.create_job_page)
         self.assertEqual(len(settings.get_help_blocks()), 3)
 
-        self.assertEqual(settings.get_job_name_error(),
-                         'This field is required.')
-        self.assertEqual(settings.get_job_start_date_error(),
-                         'This field is required.')
-        self.assertEqual(settings.get_job_end_date_error(),
-                         'This field is required.')
+        self.assertEqual(settings.get_job_name_error(), 'This field is required.')
+        self.assertEqual(settings.get_job_start_date_error(), 'This field is required.')
+        self.assertEqual(settings.get_job_end_date_error(), 'This field is required.')
 
-        # database check to ensure that job not created
+        # Check Database of job creation
         self.assertEqual(len(Job.objects.all()), 0)
```

Comment:
to be honest, I see a bigger problem - if your tests are doing E2E stuff - then you should nto even care about what the database looks like here. but, I think this is not in scope of this PR.

Reply:
Shouldn't we check for loopholes in backend if invalid instance are being created in db even after giving some error message in front?

Reply:
It's an antipattern for an E2E test. For a unit test, yes you should. In E2E - you look at direct user interactions. Cannot quote anything as from phone, but feel free to look it up.